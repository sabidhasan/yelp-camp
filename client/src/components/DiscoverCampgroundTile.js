import React from 'react'
import { Link } from 'react-router-dom'
import haversine from 'haversine'

const DiscoverCampgroundTile = (props, context) => {
  if (!props.cg) return null;
  const cgLoc = {latitude: props.cg.lat, longitude: props.cg.lon}

  return (
    <div className='DiscoverCampgroundTile'>
      {/* <a href={`/campground/${props.cg.id}`}><h2>{props.cg.name}</h2></a> */}
      <Link to={`/campground/${props.cg.id}`}>
        <h2>{props.cg.name}</h2>
      </Link>
      <p>{props.cg.region} Region</p>
      <p>{props.cg.address}</p>
      <p>~{Math.round(haversine(cgLoc, props.userLocation))} km away</p>
    </div>
  )
}

export default DiscoverCampgroundTile
