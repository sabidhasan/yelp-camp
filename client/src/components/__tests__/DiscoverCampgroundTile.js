import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DiscoverCampgroundTile from '../DiscoverCampgroundTile'

configure({ adapter: new Adapter() });

describe('<MultiCheckBox />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      cg: {lat: 20, lon: 20},
      userLocation: {latitude: 19, longitude: 19}
    }
    component = shallow(<DiscoverCampgroundTile { ...props } />)
  })

  it('If no location supplied, then return null', () => {
    props = {
      cg: null,
      userLocation: {latitude: 19, longitude: 19}
    }
    component = shallow(<DiscoverCampgroundTile { ...props } />)
    expect(component.html()).toBe(null)
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
