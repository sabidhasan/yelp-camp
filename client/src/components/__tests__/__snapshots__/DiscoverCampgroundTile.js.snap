// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiCheckBox /> LoginForm component snapshot matches what is expected 1`] = `
<div
  className="DiscoverCampgroundTile"
>
  <Link
    replace={false}
    to="/campground/undefined"
  >
    <h2 />
  </Link>
  <p>
     Region
  </p>
  <p />
  <p>
    ~
    153
     km away
  </p>
</div>
`;
